event_files/topdown.json

```diff
@@ -0,0 +1,96 @@
+    },
+    {
+        "MetricName": "TOPDOWN.BAD_SPECULATION.SLOTS_V2",
+        "BriefDescription": "TMA slots wasted due to incorrect speculations",
+        "MetricExpr": "SPEC.UOP_ISSUED - RETIRED.UOP + pipeline_width * PRD.PIPELINE_FLUSH.RECOVERY_CYCLES"
+    },
+    {
+        "EventName": "TOPDOWN.BAD_SPECULATION.CONTROL_FLOW.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots wasted due to incorrect control flow speculations.",
+        "PublicDescription": "TMA slots wasted due to incorrect control flow speculations."
+    },
+    {
+        "EventName": "TOPDOWN.BAD_SPECULATION.MEM_ORDERING.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots wasted due to memory ordering violations.",
+        "PublicDescription": "TMA slots wasted due to memory ordering violations."
+    },
+    {
+        "EventName": "TOPDOWN.BACKEND_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the lack of backend resources",
+        "PublicDescription": "TMA slots unused due to the lack of backend resources"
+    },
+    {
+        "EventName": "TOPDOWN.BACKEND_BOUND.MEMORY_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the stalls caused by load and store instructions",
+        "PublicDescription": "TMA slots unused due to the stalls caused by load and store instructions"
+    },
+    {
+        "EventName": "TOPDOWN.BACKEND_BOUND.CORE_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the non-memory stalls",
+        "PublicDescription": "TMA slots unused due to the non-memory stalls"
+    },
+    {
+        "MetricName": "TOPDOWN.BACKEND_BOUND.CORE_BOUND.SLOTS_V2",
+        "BriefDescription": "TMA slots unused due to the non-memory stalls",
+        "MetricExpr": "TOPDOWN.BACKEND_BOUND.SLOTS - TOPDOWN.BACKEND_BOUND.MEMORY_BOUND.SLOTS"
+    },
+    {
+        "EventName": "TOPDOWN.BACKEND_BOUND.MEMORY_BOUND.L1_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the stalls caused by load instructions without missing L1 data cache",
+        "PublicDescription": "TMA slots unused due to the stalls caused by load instructions without missing L1 data cache"
+    },
+    {
+        "EventName": "TOPDOWN.BACKEND_BOUND.MEMORY_BOUND.L2_BOUND.SLOTS",
+        "EventCode": "0x0",
+        "BriefDescription": "TMA slots unused due to the stalls caused by load instructions which hit L2 cache",
+        "PublicDescription": "TMA slots unused due to the stalls caused by load instructions which hit L2 cache"
+    },
```

Comment:
Worth noting that I expect these to be metrics rather than events.  Because when you're stalled waiting for data, you don't know where that data is going to come from, so you can't increment L2_BOUND, L3_BOUND, etc, until you get the data and the source information.  So instead you count cycles spent missing L2, L3, etc, and use those in a formula.  E.g., L2_BOUND = L1_MISS - L2_MISS.

Reply:
Yes makes sense, will change